### ISSUE TITLE
Product variants creation

### ISSUE THREAD

**Issue by iamchathu** [woothemes/woocommerce#32050]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Currently from the API to create a product with variants ,

First we need to create the product with all possible attributes and later we have set variant attributes.

When we integrate WooCommerce with 3rd party POS solutions when a product creates in there first main item get created and later the variants get created there is no way to know the all the variants first hand. Variants also might get added later too.

### Expected behavior

When adding variant attribute auto get updated the attributes options.

### Actual behavior

Currently variants added but not show in variant tab.

### Steps to reproduce

1. Create a variable product using api without all possible attribute set.
2. Create variants with attributes

### WordPress Environment

`
### WordPress Environment ###

WC Version: 6.3.0
REST API Version: ✔ 6.3.0
WC Blocks Version: ✔ 6.9.0
Action Scheduler Version: ✔ 3.4.0
WC Admin Version: ✔ 3.2.1
Log Directory Writable: ✔
WP Version: 5.9.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 7.4.28
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.71.0
OpenSSL/1.1.1d

SUHOSIN Installed: –
MySQL Version: 10.2.26-MariaDB-cll-lve
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 6.3.0
WC Database Prefix: woo_
Total Database Size: 7.15MB
Database Data Size: 4.35MB
Database Index Size: 2.80MB
woo_woocommerce_sessions: Data: 0.01MB + Index: 0.01MB + Engine InnoDB
woo_woocommerce_api_keys: Data: 0.01MB + Index: 0.02MB + Engine InnoDB
woo_woocommerce_attribute_taxonomies: Data: 0.01MB + Index: 0.01MB + Engine InnoDB
woo_woocommerce_downloadable_product_permissions: Data: 0.01MB + Index: 0.03MB + Engine InnoDB
woo_woocommerce_order_items: Data: 0.01MB + Index: 0.01MB + Engine InnoDB
woo_woocommerce_order_itemmeta: Data: 0.01MB + Index: 0.02MB + Engine InnoDB
woo_woocommerce_tax_rates: Data: 0.01MB + Index: 0.03MB + Engine InnoDB
woo_woocommerce_tax_rate_locations: Data: 0.01MB + Index: 0.02MB + Engine InnoDB
woo_woocommerce_shipping_zones: Data: 0.01MB + Index: 0.00MB + Engine InnoDB
woo_woocommerce_shipping_zone_locations: Data: 0.01MB + Index: 0.02MB + Engine InnoDB
woo_woocommerce_shipping_zone_methods: Data: 0.01MB + Index: 0.00MB + Engine InnoDB
woo_woocommerce_payment_tokens: Data: 0.01MB + Index: 0.01MB + Engine InnoDB
woo_woocommerce_payment_tokenmeta: Data: 0.01MB + Index: 0.02MB + Engine InnoDB
woo_woocommerce_log: Data: 0.01MB + Index: 0.01MB + Engine InnoDB
woo_actionscheduler_actions: Data: 1.52MB + Index: 1.17MB + Engine InnoDB
woo_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
woo_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
woo_actionscheduler_logs: Data: 1.52MB + Index: 0.66MB + Engine InnoDB
woo_commentmeta: Data: 0.01MB + Index: 0.02MB + Engine InnoDB
woo_comments: Data: 0.01MB + Index: 0.05MB + Engine InnoDB
woo_links: Data: 0.01MB + Index: 0.01MB + Engine InnoDB
woo_options: Data: 0.18MB + Index: 0.04MB + Engine InnoDB
woo_postmeta: Data: 0.52MB + Index: 0.06MB + Engine InnoDB
woo_posts: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
woo_termmeta: Data: 0.01MB + Index: 0.02MB + Engine InnoDB
woo_terms: Data: 0.01MB + Index: 0.02MB + Engine InnoDB
woo_term_relationships: Data: 0.01MB + Index: 0.01MB + Engine InnoDB
woo_term_taxonomy: Data: 0.01MB + Index: 0.02MB + Engine InnoDB
woo_usermeta: Data: 0.01MB + Index: 0.02MB + Engine InnoDB
woo_users: Data: 0.01MB + Index: 0.02MB + Engine InnoDB
woo_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
woo_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
woo_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
woo_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
woo_wc_download_log: Data: 0.01MB + Index: 0.02MB + Engine InnoDB
woo_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
woo_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
woo_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
woo_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
woo_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
woo_wc_product_meta_lookup: Data: 0.01MB + Index: 0.05MB + Engine InnoDB
woo_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
woo_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
woo_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
woo_wc_webhooks: Data: 0.01MB + Index: 0.01MB + Engine InnoDB
woo_yith_wcwl: Data: 0.01MB + Index: 0.01MB + Engine InnoDB
woo_yith_wcwl_lists: Data: 0.01MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
page: 7
post: 2
product: 13
product_variation: 31
shop_order: 8

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ❌Error messages should not be shown to visitors.

### Active Plugins (3) ###

WooCommerce: by Automattic – 6.3.0
WP Blog Post Layouts: by CodeVibrant – 1.1.0
YITH WooCommerce Wishlist: by YITH – 3.0.23

### Inactive Plugins (3) ###

Akismet Anti-Spam: by Automattic – 4.2.2
Hello Dolly: by Matt Mullenweg – 1.7.2
WP Maintenance Mode & Coming Soon: by Themeisle – 2.4.3

### Settings ###

API Enabled: –
Force SSL: –
Currency: LKR (රු)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #6 - /
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Easy Mart
Version: 1.0.5
Author URL: https://codevibrant.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 3,905
Oldest: 2022-03-09 06:48:49 +0000
Newest: 2022-03-10 04:29:24 +0000

### Status report information ###

Generated at: 2022-03-10 10:05:22 +05:30
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by claudiosanches** [comment]
Hi @iamchathu,

While our goal is to address all the issues reported in this repository, GitHub should be treated as a place to report confirmed bugs only.

The type of issue you submitted looks like a custom code related question. Due to the number of bug reports being submitted, we are prioritizing work on them and are not able to attend to custom code questions. 

The best place to get help further is in the [WooCommerce Community Slack](https://woocommerce.com/community-slack/) in the `#developers` channel.

If you confirm the bug, please open a new issue and provide us with clear steps to reproduce it.

I am closing this issue for now. If I missed something, please leave a comment and we will take a second look.

**Comment by woocommercebot** [comment]
Hi @iamchathu,

  While our goal is to address all the issues reported in this repository,   GitHub should be treated as a place to report confirmed bugs only.

  The type of issue you submitted looks like a support request which may or may not reveal a bug once proper   troubleshooting is done.  In order to confirm the bug, please follow one of the steps below:

  - Review [WooCommerce Self-Service Guide](https://docs.woocommerce.com/document/woocommerce-self-service-guide/)   to see if the solutions listed there apply to your case;
  - If you are a paying customer of WooCommerce, contact WooCommerce support by   [opening a ticket or starting a live chat](https://woocommerce.com/contact-us/);
  - Make a post on [WooCommerce community forum](https://wordpress.org/support/plugin/woocommerce/)

  If you confirm the bug, please provide us with clear steps to reproduce it.

  We are closing this issue for now as it seems to be a support request and not a bug.   If we missed something, please leave a comment and we will take a second look.

**Comment by iamchathu** [comment]
> Hi @iamchathu,
> 
> While our goal is to address all the issues reported in this repository, GitHub should be treated as a place to report confirmed bugs only.
> 
> The type of issue you submitted looks like a custom code related question. Due to the number of bug reports being submitted, we are prioritizing work on them and are not able to attend to custom code questions.
> 
> The best place to get help further is in the [WooCommerce Community Slack](https://woocommerce.com/community-slack/) in the `#developers` channel.
> 
> If you confirm the bug, please open a new issue and provide us with clear steps to reproduce it.
> 
> I am closing this issue for now. If I missed something, please leave a comment and we will take a second look.

It's not a custom code related issue. This is a WooCommerce API related issue. Which is use by a custom code.

**Comment by claudiosanches** [comment]
> First we need to create the product with all possible attributes and later we have set variant attributes.

This is the expected behavior for the REST API, it's required to create the main product first with all attributes, this is also how works in the admin interface when adding products. It's possible for you to write a middleware to handle this kind of data from your POS and properly handle it to sent to the WooCommerce REST API.

I understand if it doesn't fit all your needs, you could instead file a enhancement request instead of a bug report, so you could request for this feature.
